packages/components/src/internal/components/lineage/models.ts

```diff
@@ -974,36 +967,220 @@ export function isAliquotNodeCollection(node: LineageNodeCollection | LineageNod
+                    const edgeId = makeEdgeId(combinedNode.id, existingEdge.to);
+                    // console.log("  ".repeat(depth) + "from matches. creating new edge: " + edgeId);
+                    visEdges[edgeId] = {
+                        id: edgeId,
+                        from: combinedNode.id,
+                        to: existingEdge.to,
+                    };
+                } else if (existingEdge.to === existingCombinedId) {
+                    // create a new edge from the combined node from the existing basic node
+                    const edgeId = makeEdgeId(existingEdge.from, combinedNode.id);
+                    // console.log("  ".repeat(depth) + "to matches. creating new edge: " + edgeId);
+                    visEdges[edgeId] = {
+                        id: edgeId,
+                        from: existingEdge.from,
+                        to: combinedNode.id,
+                    };
+                }
+            }
+        }
+    }
+
+    // create association from the node within the combined node to the combined node
+    if (!nodesInCombinedNode.hasOwnProperty(edge.lsid)) {
+        nodesInCombinedNode[edge.lsid] = [];
+    }
+    nodesInCombinedNode[edge.lsid].push(combinedNode.id);
+}
+
+/** Applies the {@link LineageGroupingOptions#combineSize} logic to the graph. */
+function applyCombineSize(
+    lsid: string,
+    edges: List<LineageLink>,
+    nodes: LineageNodeMap,
+    options: LineageOptions,
+    dir: LINEAGE_DIRECTIONS,
+    visEdges: EdgeMap,
+    visNodes: VisNodeMap,
+    nodesInCombinedNode: { [key: string]: string[] }
+): LineageNode[] {
+    const combinedLineageNodes: LineageNode[] = [];
+    let combinedAliquotNode: VisGraphCombinedNode;
+    let combinedNonAliquotNode: VisGraphCombinedNode;
+
+    const node = nodes[lsid];
+    const { aliquotEdges, nonAliquotEdges } = edges.reduce(
+        (acc, e) => {
+            if (nodes[e.lsid].materialLineageType === 'Aliquot') {
+                acc.aliquotEdges.push(e);
+            } else {
+                acc.nonAliquotEdges.push(e);
+            }
+            return acc;
+        },
+        { aliquotEdges: [] as LineageLink[], nonAliquotEdges: [] as LineageLink[] }
+    );
+
+    if (aliquotEdges.length > 1) {
+        const aliquotNodes = aliquotEdges.map(e => nodes[e.lsid]);
+
+        combinedAliquotNode = createCombinedVisNode(aliquotNodes, options, node.name);
+        visNodes[combinedAliquotNode.id] = combinedAliquotNode;
+        combinedLineageNodes.push(createCombinedLineageNode(combinedAliquotNode));
+
+        // create a VisGraph Edge from the current node to the new combined node
+        // as well as an edge for each of the combined nodes that one of the edge target nodes may belong to.
+        addEdges(lsid, combinedAliquotNode.id, visEdges, List(aliquotEdges), nodesInCombinedNode, dir);
+    }
+
+    if (edges.size - aliquotEdges.length >= options.grouping.combineSize) {
```

Comment:
edges.size - aliquotEdges.length = nonAliquotEdiges, right?

Reply:
Agreed. Hadn't changed this logic from before (this PR) but I'll update it now.